### DISCUSSION TITLE
TRANSMISSION PERMISSION DENIED(13) WRITING IN SHARED FOLDER

### DISCUSSION THREAD

**Discussion by RafTube** [transmission/transmission#8363] [Support]
### What is the issue?

Im trying to download a torrent using a VM Ubuntu 22.04 in a shared folder from my Windows (host) to Linux but after a fraction of a second it throws an error message saying it (the transmission) doesnt have the right permissions to write on said folder. I feel like ive tried everything. I tried adding the "debian-transmission" user to root or vboxsf groups as some forums suggested but the user isnt found. I tried using chmod on the shared folder. It appeared to work but didnt. The worst part is im fairly new to Linux so Im kinda lost in here. Me myself as a user can rwx on the folder but the app (which should be running under the users allowed permissions i think) cant. Im lost in 23 different forums not knowing what to do. Any help is appreciated 

### Which application of Transmission?

Qt app on Linux, BSD, etc.

**Comment by killemov** [comment]
So it's Windows -> VM -> Ubuntu -> Transmission? (Transmission in docker perhaps?)

ANY of these layers could have write restrictions. As a test you could open a shell in Ubuntu as user "debian-transmission" and write a file in the targeted downloads directory. There is a good chance that you may have incorrectly configured the downloads directory so check the settings.json as well.

**Comment by RafTube** [comment]
> So it's Windows -> VM -> Ubuntu -> Transmission? (Transmission in docker perhaps?)
> 
> ANY of these layers could have write restrictions. As a test you could open a shell in Ubuntu as user "debian-transmission" and write a file in the targeted downloads directory. There is a good chance that you may have incorrectly configured the downloads directory so check the settings.json as well.



How exactly can I do that? As I said im very new to Linux but i want the extra security while torrenting. The sequence you typed Win->VM->etc is correct. What  confuses me most is that I the user can write there but transmission cant. How is that possible?

**Comment by RafTube** [comment]
I also found out that /_**etc/systemd/system/transmission-daemon.service.d/override.conf.**_  and /_**etc/systemd/system/multi-user.target.wants/transmission-daemon.service**_ dont even exist in my directory which they should be

**Comment by killemov** [comment]
Ok. So no Docker, which excludes a whole possible world of complications.

> What confuses me most is that I the user can write there but transmission cant. How is that possible?

Because `raftube` != `debian-transmission`. Make sure the downloads directory and its parents have the correct write access. You could change user+/group ownership of these to `debian-transmission`.

> I also found out that /_**etc/systemd/system/transmission-daemon.service.d/override.conf.**_ and /_**etc/systemd/system/multi-user.target.wants/transmission-daemon.service**_ dont even exist in my directory which they should be

Did you install transmission in a regular way? (apt or similar) Because then at least the service file should be available.

**Comment by RafTube** [comment]
> Ok. So no Docker, which excludes a whole possible world of complications.
> 
> > What confuses me most is that I the user can write there but transmission cant. How is that possible?
> 
> Because `raftube` != `debian-transmission`. Make sure the downloads directory and its parents have the correct write access. You could change user+/group ownership of these to `debian-transmission`.
> 
> > I also found out that /_**etc/systemd/system/transmission-daemon.service.d/override.conf.**_ and /_**etc/systemd/system/multi-user.target.wants/transmission-daemon.service**_ dont even exist in my directory which they should be
> 
> Did you install transmission in a regular way? (apt or similar) Because then at least the service file should be available.

I installed transmission through App Center, and as I said before the user **_debian-transmission_** doesnt exist. I tried running _**sudo chown -R debian-transmission /home/username/sharedfolder**_ but it responded with **_chown: invalid user: ‘debian-transmission’_**

**Comment by killemov** [comment]
A quick search lead me to this: https://askubuntu.com/questions/261252/how-do-i-change-the-user-transmission-runs-under
Apparently Ubuntu deviates from standard Debian by running as you. You could create both user and group `debian-transmission` as a system user and then set the proper access rights and then run transmission under those privileges.
